### ISSUE TITLE
Package Maintenance for TF-2.0 with Contrib Module Sunset

### ISSUE THREAD

**Issue by Uiuran** [openai/gradient-checkpointing#48]
Iam wondering if i start to use the package, what seems to be an excel option to do models from scratch (including contrib to the package if i need to modify it to fit my model archs), do you plan to do maintenance for TF-2.0 ? 

graph_editor gonna be erased from the framework since until now there is no proposal for maintenance.

**Comment by yaroslavvb** [comment]
no, did not plan to update it for TF 2.0 -- I'm assuming TF would come up with an official solution given how important this is

**Comment by Uiuran** [comment]
Agree with you in relation to the importance, but it does not seem to be a priority (at last i don't see anything in mvt right now). Maybe we can do Improvement Proposal or Addon Proposal.

**Comment by den-run-ai** [comment]
There is partially open-source/free LMS from IBM PowerAI for Intel/Power CPUs & NVidia GPUs, also native checkpointing in PyTorch:

https://pytorch.org/docs/stable/checkpoint.html

**Comment by Uiuran** [comment]
Since the problem is not trivial and may not be a priority in TensorFlow, one solution may be clone graph_editor to a branch of this project, make the necessary changes to -> tf 2.0  and merge it as a unique project.
I would like to contribute in the near future for this.
Any official update in TF Team pls update/ping here.

**Comment by shsshs** [comment]
TF 2.0 does now have an experimental graph editor included (if i understood correctly)

**Comment by Uiuran** [comment]
I cant see it. You probably is talking about Grappler or Autograph, the first works as a comp. graph optimizator, the other does graph building from the new TF 2.0 API, which is much more like pythonic code. 
Afaik, none of them is a module structured to do graph editing.  Is there any other ? or am i wrong in relation  to these ?

**Comment by shsshs** [comment]
https://github.com/tensorflow/tensorflow/issues/30418

was refering to tf.recompute_grad as suggested at the bottom. but i guess this is no solution for just using this repository.

**Comment by shsshs** [comment]
and there is this:
https://github.com/CODAIT/graph_def_editor

but it does not look finished and - given the speed at which tf 2 evolves - it will most likely stay like this for a while.

**Comment by Uiuran** [comment]
> and there is this:
> https://github.com/CODAIT/graph_def_editor
> 
> but it does not look finished and - given the speed at which tf 2 evolves - it will most likely stay like this for a while.

This seems to be an unofficial extension, however it seems to be the same purpose of this question, hopes they can make the extension proposal, closing the question for while.

**Comment by nouiz** [comment]
I just found this other packages that seem to work on TF 2.0:
https://checkmateai.github.io/ I didn't try it.

**Comment by yaroslavvb** [comment]
How do you know it's TF 2.0? I can't find any code by following that link

**Comment by Uiuran** [comment]
good question, there is only  a readme.md in the repository.
